File: tutorials/conversion_obj_step/conversion_obj_step.py
#!/usr/bin/env python3


from kittycad.api.file import create_file_conversion
from kittycad.client import ClientFromEnv
from kittycad.models.error import Error
from kittycad.models.file_conversion import FileConversion
from kittycad.models.file_export_format import FileExportFormat
from kittycad.models.file_import_format import FileImportFormat

# Create a new client with your token parsed from the environment variable:
#   KITTYCAD_API_TOKEN.
client = ClientFromEnv(timeout=500, verify_ssl=True)

# Convert a file from OBJ to STEP.
# Read in the contents of the file.
file = open("./ORIGINALVOXEL-3.obj", "rb")
content = file.read()
file.close()

result = create_file_conversion.sync(
    client=client,
    body=content,
    src_format=FileImportFormat.OBJ,
    output_format=FileExportFormat.STEP,
)

if isinstance(result, Error) or result is None:
    raise Exception("There was a problem")

fc: FileConversion = result

print(f"File conversion id: {fc.id}")
print(f"File conversion status: {fc.status}")

if fc.outputs.length != 1:
    raise Exception("Expected one output file")

# LITTERBOX-START-NON-EDITABLE-SECTION
for _, output in fc.outputs.items():
    output_file_path = "./output.step"
    print(f"Saving output to {output_file_path}")
    output_file = open(output_file_path, "wb")
    output_file.write(output.get_decoded())
    output_file.close()

File: tutorials/conversion_obj_step/conversion_obj_step.go
package main

import (
	"fmt"
	"os"

	"github.com/kittycad/kittycad.go"
)

func main() {
	// Create a new client with your token parsed from the environment variable:
	// KITTYCAD_API_TOKEN.
	client, err := kittycad.NewClientFromEnv("your apps user agent")
	if err != nil {
		panic(err)
	}

	fileBytes, _ := os.ReadFile("./ORIGINALVOXEL-3.obj")

	fc, err := client.File.CreateConversion(
		kittycad.FileExportFormatStep,
		kittycad.FileImportFormatObj,
		fileBytes,
	)
	if err != nil {
		panic(err)
	}

	fmt.Println("File conversion id: ", fc.ID)
	fmt.Println("File conversion status: ", fc.Status)

	output_file_path := "./output.step"
	fmt.Println("Saving output to ", output_file_path)

	output, err := os.Create(output_file_path)
	if err != nil {
		panic(err)
	}
	defer output.Close()

	if len(fc.Outputs) != 1 {
		panic("Expected 1 output file")
	}

	for _, contents := range fc.Outputs {
		if _, err := output.Write(contents.Inner); err != nil {
			panic(err)
		}
	}

	if err := output.Sync(); err != nil {
		panic(err)
	}
}
